include/arch/x86/arch.h

```diff
@@ -453,6 +453,15 @@ static ALWAYS_INLINE void z_arch_irq_unlock(unsigned int key)
 	z_do_irq_unlock();
 }
 
+/**
```

Comment:
IMHO this comment should be before every z_arch_irq_unlocked() definition, as the function name might be misleading.
Have you considered different name? For example z_arch_irq_was_unlocked() ?